src/server/svr_jobfunc.c

```diff
@@ -6468,37 +6467,27 @@ return_missing_resources(char *chunk, char *res_list)
 	if (rc != 0) {
 		snprintf(log_buffer,  sizeof(log_buffer)-1,
 			"bad parse of %s", chunk_dup);
-		log_err(-1, id, log_buffer);
+		log_err(-1, __func__, log_buffer);
 		free(chunk_dup);
 		return (NULL);
 	}
 	ret_buf[0] = '\0';
-	for (l=0; l < snelma; ++l) {
+	for (l = 0; l < snelma; ++l) {
 		if (!in_string_list(skv[l].kv_keyw, ',', res_list)) {
 			/* not seen in exec_vnode */
 			buf_sz = sizeof(ret_buf);
 			if (ret_buf[0] == '\0') {
-				snprintf(ret_buf, buf_sz-1,
-				  	"%s=%s",
-				   skv[l].kv_keyw,
-				   skv[l].kv_val);
+				snprintf(ret_buf, buf_sz,
+				  	"%s=%s", skv[l].kv_keyw, skv[l].kv_val);
 			} else {
 				buf_len = strlen(ret_buf);
-				strncat(ret_buf,
-				   ":",
-				   buf_sz-buf_len-1);
+				strncat(ret_buf, ":", buf_sz-buf_len-1);
 				buf_len += 1;
-				strncat(ret_buf,
-				   skv[l].kv_keyw,
-				   buf_sz-buf_len-1);
+				strncat(ret_buf, skv[l].kv_keyw, buf_sz-buf_len-1);
 				buf_len = strlen(ret_buf);
-				strncat(ret_buf,
-				   "=",
-				   buf_sz-buf_len-1);
+				strncat(ret_buf, "=", buf_sz-buf_len-1);
 				buf_len += 1;
-				strncat(ret_buf,
-				   skv[l].kv_val,
-				   buf_sz-buf_len-1);
+				strncat(ret_buf, skv[l].kv_val, buf_sz-buf_len-1);
```

Comment:
Only the first character of ret_buf gets initialized to '\0'.  You need to handle the case when strncat() doesn't terminate the string.